cryus-sasl: cleanup, fix w/musl (although not at all musl-specific)

dtzWill opened the pull request:
musl is fixed by disabling update of config.{guess,sub}.


Let's check how this fares on Aarch64, since that platform (I think?)
also automatically updates these files by default.

Fixes #47105.


<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

dtzWill committed:
cryus-sasl: cleanup, fix w/musl (although not at all musl-specific)

musl is fixed by disabling update of config.{guess,sub}.

dtzWill commented:
Hopefully 2.1.27 is soon and is much less broken regarding autoconf/libtool bits.  Looks like we're in the minority for not having a big stack of patches for this already.
(the LFS patch currently applied on FreeBSD has some fixes for this, FWIW)

Mic92 commented:
staging seems appropriate: https://gist.github.com/GrahamcOfBorg/5308396b96f7eb5c758a7fa7c5193585

dtzWill commented:
Good call, and sorry for the review-invite spam after rebase.

matthewbauer commented on pkgs/development/libraries/cyrus-sasl/default.nix:
```diff
@@ -36,15 +36,15 @@ stdenv.mkDerivation rec {
-  '';
+  # Avoid triggering regenerating using broken autoconf/libtool bits.
+  # (many distributions carry patches to remove/replace, but this works for now)
+  dontUpdateAutotoolsGnuConfigScripts = true;
```
I wonder why this breaks things?

matthewbauer replied:
Maybe set it to null if not musl.

matthewbauer replied:
Yeah makes sense. I wonder if it just needs to regenerate some files though? Changing the config.* files should be harmless.

dtzWill committed:
cyrus-sasl: set to null on non-musl per reviewer suggestion

Unclear what the problem is exactly regarding regenerating files,
so this makes the change only impact build configs known to need it.

dtzWill commented:
Alright I'll just set to null on non-musl.  Thanks!
